lib/src/util/posix/dynamic_library.cc

```diff
@@ -41,10 +41,10 @@ namespace facter { namespace util {
         close();
 
         // Don't actually perform a load to determine if it is already loaded
-#ifdef _AIX
+#if defined(_AIX) || !defined(RTLD_NOLOAD)
```

Comment:
Can probably remove the `_AIX` check now, as `RTLD_NOLOAD` won't be defined in either case.

I think we can probably remove the usage of `RTLD_NOLOAD` entirely, as we search for exported ruby symbols in POSIX and for loaded modules by name on Windows to find the existing library before attempting to load it.  This code existed before those checks were made.